packages/react-native/ReactCommon/react/renderer/graphics/platform/ios/react/renderer/graphics/HostPlatformColor.mm

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#import "HostPlatformColor.h"
+
+#import <Foundation/Foundation.h>
+#import <React/RCTConvert.h>
+#import <UIKit/UIKit.h>
+#include <react/utils/ManagedObjectWrapper.h>
+#include <string>
+
+using namespace facebook::react;
+
+NS_ASSUME_NONNULL_BEGIN
+
+namespace facebook::react {
+
+inline UIColor *_Nullable RCTUIColorFromDynamicColor(const facebook::react::DynamicColor &dynamicColor)
+{
+  int32_t light = dynamicColor.lightColor;
+  int32_t dark = dynamicColor.darkColor;
+  int32_t highContrastLight = dynamicColor.highContrastLightColor;
+  int32_t highContrastDark = dynamicColor.highContrastDarkColor;
+
+  UIColor *lightColor = [RCTConvert UIColor:@(light)];
+  UIColor *darkColor = [RCTConvert UIColor:@(dark)];
+  UIColor *highContrastLightColor = [RCTConvert UIColor:@(highContrastLight)];
+  UIColor *highContrastDarkColor = [RCTConvert UIColor:@(highContrastDark)];
+
+  if (lightColor != nil && darkColor != nil) {
+    UIColor *color = [UIColor colorWithDynamicProvider:^UIColor *_Nonnull(UITraitCollection *_Nonnull collection) {
+      if (collection.userInterfaceStyle == UIUserInterfaceStyleDark) {
+        if (collection.accessibilityContrast == UIAccessibilityContrastHigh && highContrastDarkColor != nil) {
+          return highContrastDarkColor;
+        } else {
+          return darkColor;
+        }
+      } else {
+        if (collection.accessibilityContrast == UIAccessibilityContrastHigh && highContrastLightColor != nil) {
+          return highContrastLightColor;
+        } else {
+          return lightColor;
+        }
+      }
+    }];
+    return color;
+  } else {
+    return nil;
+  }
+
+  return nil;
+}
+
+static inline int32_t _ColorFromUIColor(UIColor *color)
+{
+  float ratio = 255;
+  CGFloat rgba[4];
+  [color getRed:&rgba[0] green:&rgba[1] blue:&rgba[2] alpha:&rgba[3]];
+  return ((int32_t)round((float)rgba[3] * ratio) & 0xff) << 24 | ((int)round((float)rgba[0] * ratio) & 0xff) << 16 |
+      ((int)round((float)rgba[1] * ratio) & 0xff) << 8 | ((int)round((float)rgba[2] * ratio) & 0xff);
```

Comment:
Re-use getColorComponents?

Reply:
Seems I have no idea about how to re-use, can you please explain more?

Reply:
Never mind, this is going in the opposite direction of `getColorComponents`